### PR TITLE
EPMDJ-1176. Notifications implementation

### PR DESCRIPTION
**Pull request by s0larm00n**

### CODE REVIEW

File: src/main/kotlin/com/epam/drill/endpoints/AgentManager.kt [hunk 1]
```diff
@@ -35,7 +38,16 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
...
-                    }.apply { buildVersions = SizedCollection(buildVersions.toSet() + this) }
+                    }.apply {
+                        buildVersions = SizedCollection(buildVersions.toSet() + this)
+                        notificationsManager.save(
+                            agentId,
+                            agentInfoDb.name,
+                            NotificationType.BUILD,
+                            notificationsManager.buildArrivedMessage(pBuildVersion)
+                        )
+                        topicResolver.sendToAllSubscribed("/notifications")
```
Comment (IgorKey) [inline src/main/kotlin/com/epam/drill/endpoints/AgentManager.kt]: may be incapsulate this func into manager?

Comment (s0larm00n) [reply src/main/kotlin/com/epam/drill/endpoints/AgentManager.kt]: unfortunately, this is not the only entity using this method

File: src/main/kotlin/com/epam/drill/util/NotificationsManager.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            NotificationStatus.UNREAD,
+            type,
+            message
+        )
+    }
+
+    fun read(id: String): Boolean = notifications[id]?.let {
+        val readNotification = it.copy(status = NotificationStatus.READ)
+        notifications[id] = readNotification
+        notifications[id]!!.status == NotificationStatus.READ
```
Comment (IgorKey) [inline src/main/kotlin/com/epam/drill/util/NotificationsManager.kt]: wtf :D

File: src/main/kotlin/com/epam/drill/endpoints/agent/TopicResolver.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import kotlinx.serialization.*
+import org.kodein.di.*
+import org.kodein.di.generic.*
+
+class TopicResolver(override val kodein: Kodein) : KodeinAware {
+    private val app: Application by instance()
+    private val wsTopic: WsTopic by instance()
+    private val sessionStorage: SessionStorage by instance()
+
+    @UseExperimental(ImplicitReflectionSerializer::class)
```
Comment (IgorKey) [inline src/main/kotlin/com/epam/drill/endpoints/agent/TopicResolver.kt]: is this still required?

Comment (s0larm00n) [reply src/main/kotlin/com/epam/drill/endpoints/agent/TopicResolver.kt]: looks like nope

**Pull request merged**